assembly/src/assembler/context.rs

```diff
@@ -3,7 +3,7 @@ use super::{
     NamedProcedure, Procedure, ProcedureCache, ProcedureId, ProcedureName, RpoDigest, ToString,
     Vec,
 };
-use crate::ast::{ModuleAst, ProgramAst};
+use crate::ast::{event, Level, ModuleAst, ProgramAst};
```

Comment:
nit: it is odd to import the `tracing::event` macro from the `ast` module. Maybe import it directly?